tests/data/MerchFactory.ts

```diff
@@ -18,6 +19,13 @@ export class MerchFactory {
     return FactoryUtils.create(n, MerchFactory.fakeOption);
   }
 
+  public static createOptionMetadataOfSameType(n: number): MerchItemOptionMetadata[] {
```

Comment:
Nit: `createOptionMetadata`, since this method would only create valid metadata.